### PR TITLE
Updates to Branch Manager Handbook

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

This PR adds a few changes and updates to the branch management handbook:

* Update Google Cloud SDK authentication for `krel history`
* Reformat Release Issue title structure
* Remove `cip-mm` requirements
* Remove `testgridshot` script references and change instructions to `krel testgridshot`
* Rebuild TOC

Signed-off-by: Adolfo García Veytia (Puerco) <adolfo.garcia@uservers.net>

#### Which issue(s) this PR fixes:
None

### CODE REVIEW

File: release-engineering/role-handbooks/branch-manager.md [hunk 1]
```diff
@@ -206,47 +208,77 @@ There are more examples of the release process under the [References](#reference
...
-./testgridshot <release>
 ```
+krel testgridshot --branch 1.20 --github-issue 12345 --states=FLAKY
+```
+
+Once the script generates the Markdown table, post it as a comment on the created issue. You can take a look at the [following comment](https://github.com/kubernetes/sig-release/issues/1249#issuecomment-696702503) as an example.
+
+#### Adding data about the Cloud Build Jobs
+
+When running a release cut, you should open a thread in the [#release-management]
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:245]: `[#release-management]` is not linked to anything

**Review by saschagrunert — APPROVED** [review]
Thank you for the update @puerco! :pray: Just one nit, otherwise LGTM

File: release-engineering/role-handbooks/branch-manager.md (hunk 1 above)
Comment (puerco) [reply release-engineering/role-handbooks/branch-manager.md]: Done ;:)

**Review by hasheddan — APPROVED** [review]
/lgtm

**Pull request merged**